pkg/webhooks/server.go

```diff
@@ -126,13 +125,13 @@ func (s *server) Cleanup() <-chan struct{} {
 }
 
 func (s *server) cleanup(ctx context.Context) {
-	cleanupKyvernoResource := s.webhookRegister.ShouldCleanupKyvernoResource()
+	// cleanupKyvernoResource := s.webhookRegister.ShouldCleanupKyvernoResource()
 
-	var wg sync.WaitGroup
-	wg.Add(2)
-	go s.webhookRegister.Remove(cleanupKyvernoResource, &wg)
-	go s.webhookRegister.ResetPolicyStatus(cleanupKyvernoResource, &wg)
-	wg.Wait()
+	// var wg sync.WaitGroup
+	// wg.Add(2)
+	// go s.webhookRegister.Remove(cleanupKyvernoResource, &wg)
+	// go s.webhookRegister.ResetPolicyStatus(cleanupKyvernoResource, &wg)
+	// wg.Wait()
```

Comment:
What are we supposed to cleanup here ?

Reply:
All Kyverno webhook configurations and the lease objects.

Reply:
The kyvernopre lease right ?
Why shall we delete the webhook configurations here ?

Reply:
>The kyvernopre lease right ?

All kyverno leases.

Reply:
>Why shall we delete the webhook configurations here ?
We don't even checker we are a leader...

This check guarantees that the Kyverno deployment is removed, we need to clean up all webhooks otherwise the requests hang.
```
s.webhookRegister.ShouldCleanupKyvernoResource()
```

Reply:
Ok, I will add it back. Not sure if we need it though, would be better to react to deployment changes (scaled to 0, or deleted should be the right place to do it).
And deleting the leases is useless, leaderelection takes care of releasing the lease if the holder is not live.

Reply:
Going to add this logic in the controller and check if it works.

Reply:
>And deleting the leases is useless, leaderelection takes care of releasing the lease if the holder is not live.

Right. But it's always a good practice to do the garbage collecting thing if resources are no longer in use.

Reply:
> But it's always a good practice to do the garbage collecting thing if resources are no longer in use.

I would expect the user to delete the namespace TBH.

Reply:
The `helm uinstall` won't do that. We cannot rely on users for cleanup.

Reply:
We should at least clean up webhook configurations upon deletion/scale down to zero.

@eddycharly - are you adding it back in this PR?

Reply:
@realshuting I added deletion of leases/webhook configs in this PR.

Reply:
Thanks!